### ISSUE TITLE
mod_full_brace_if_chain=true breaks initializer-list parameter in function call within IF-ELSE

### ISSUE THREAD

**Issue by jknightmmcs** [uncrustify/uncrustify#2322]
Hey guys :)

For some reason, my uncrustify **0.68.1 (Windows 10 1607 LTSB [14393])** crashes when I try to call it with **-p debug.txt** parameter, so I can't include it.

Basically, here is the problem (smallest example I could come up with):

take this:
```
int main()
{
    if (foo()) bar();
    else if (baz( { rick, morty })) anime();
}
```
then use uncrustify on it, and it becomes this:

```
int main()
{
    if (foo())
        bar();
    else if (baz( rick, morty))
        anime();
}
```
Those braces are removed for some reason, and they shouldn't be.
**mod_full_brace_if_chain=false** fixes it.

My full uncrustify config: [link](https://pastebin.com/cfRHtBr5)

**Comment by guy-maurel** [comment]
1. use of -p p.txt
The current git version doesn't crash.
  https://github.com/uncrustify/uncrustify
Please test it and report!

**Comment by guy-maurel** [comment]
2.
You need to use the option:
 ` nl_if_leave_one_liners=true`

**Comment by jknightmmcs** [comment]
**Uncrustify-0.69.0_f, built in Debian**, -p option worked. With this version issue is still in place.

[Here is the link.](https://pastebin.com/F0MG6RQj)

What's for second message, I think that there should be ZERO amount of option combinations that break existing code. Adding this line should NOT be required.

Also, adding this line doesn't fix the issue (at least with 0.69.0_f).

One and only thing that sometimes breaks code I can think of being worthy to keep in configuration is clang-format's headers automatic reaarrangement.

**Comment by jknightmmcs** [comment]
Just to bump the thread =3 

In the code (first message) there is baz(std::pair<T1, T2>) call that contains initializer-list structure {rick, morty}. Using uncrustify on that code removes curly braces, which transforms the code to a two-parameter call - baz(T1, T2). Please, look into it and fix the issue.
